### DISCUSSION TITLE
How do i reset range slider with a reset button

### DISCUSSION THREAD

**Discussion by snhnaik8** [chakra-ui/chakra-ui#7598] [Q&A]
### Description

I have following child  component which is a part of form with reset button , on click of it I need range slider position back to default values.  On click of  reset button the state value changes however the position rangeslider thumb doesn't change

 <FormControl sx={styles.formControl}>
            <FormLabel sx={styles.legend}>Year</FormLabel>

            <Box px={6}>
                <RangeSlider aria-label={['min', 'max']} defaultValue={[2010, 2022]} min={2010} max={2022} step={1} onChangeEnd={(val) => handleChangePrice(val)}>
                    <RangeSliderTrack bg='lightgray'>
                        <RangeSliderFilledTrack bg='gray5' />
                    </RangeSliderTrack>
                    <RangeSliderThumb boxSize={10} index={0}>
                        <Box color='gray5' as={BiChevronLeftCircle} boxSize={10} />
                    </RangeSliderThumb>
                    <RangeSliderThumb boxSize={10} index={1}>
                        <Box color='gray5' as={BiChevronRightCircle} boxSize={10} />
                    </RangeSliderThumb>
                </RangeSlider>
            </Box>
            <Flex>
                <Box px='4'>
                    {minYear}
                </Box>
                <Spacer />
                <Box px='4'>
                    {maxYear}
                </Box>
            </Flex>
        </FormControl>

### Link to Reproduction

dont have link

### Steps to reproduce

2. Click on '...'
3. Scroll down to '...'

### Chakra UI Version

2.2.4

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux

**Comment by qasimalyas** [comment]
Do you think you can create an example on codesandbox or something. Would certainly help in debugging and to see whats going on.

**Comment by xiayang-xy** [comment]
I hava the same question😫

**Comment by dvartic** [comment]
Hello. Not sure if you guys still having this issue, since it's old. But to do what you want you need to make your Sliders a complete controlled component, meaning it's value needs to be paired to a React state, it's not enough to use `onChangeEnd`, that will only call the provided function when the user stops moving the slider, but it won't pair it's value to a state. You need to use `value` and `onChange` props.

If needed, you can also combine `value`, `onChange` and `onChangeEnd`, separate the sliders into independent components, so that you can handle current state separate from onChangeEnd state. This prevents rerenders, if you want to have a controlled Slider, but the actual state value you use for your logic is only the final value.

See following code from one of my projects:

```tsx
interface FilterValues {
    grupos: GrupoOposicion[];
    plazas: number[];
    aspirantes: number[];
    aspirantesPlaza: number[];
    experienceAverage: number[];
}
interface Props {
    filterValues: FilterValues;
    filterState: FilterValues;
    handleGrupoChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
    handlePlazasChange: (values: number[]) => void;
    handleAspirantesChange: (values: number[]) => void;
    handleAspirantesPlazaChange: (values: number[]) => void;
    handleExperienceChange: (values: number[]) => void;
    postRangeSliderState: () => void;
}

export function AccordionFilters({
    filterValues,
    filterState,
    handleGrupoChange,
    handlePlazasChange,
    handleAspirantesChange,
    handleAspirantesPlazaChange,
    handleExperienceChange,
    postRangeSliderState,
}: Props) {
    return (
        <RangeSlider
            min={filterValues.plazas[0]}
            max={filterValues.plazas[1]}
            value={filterState.plazas}
            onChange={handlePlazasChange}
            onChangeEnd={postRangeSliderState}
        >
            <RangeSliderTrack>
                <RangeSliderFilledTrack />
            </RangeSliderTrack>
            <RangeSliderThumb index={0} />
            <RangeSliderThumb index={1} />
        </RangeSlider>
    );
}

```

**Reply by mauro-mascia** [reply]
> Hello. Not sure if you guys still having this issue, since it's old. But to do what you want you need to make your Sliders a complete controlled component, meaning it's value needs to be paired to a React state, it's not enough to use `onChangeEnd`, that will only call the provided function when the user stops moving the slider, but it won't pair it's value to a state. You need to use `value` and `onChange` props.
> 
> If needed, you can also combine `value`, `onChange` and `onChangeEnd`, separate the sliders into independent components, so that you can handle current state separate from onChangeEnd state. This prevents rerenders, if you want to have a controlled Slider, but the actual state value you use for your logic is only the final value.

Thanks it works
